espflash/src/cli/monitor/mod.rs

```diff
@@ -102,7 +108,16 @@ pub fn monitor_with<L: InputParser>(
             err => err,
         }?;
 
-        parser.feed(&buff[0..read_count], &mut stdout);
+        match log_format {
```

Comment:
I'm not sure why I didn't notice this but this isn't correct. The defmt parser is stateful, we can't just create a new one every time we have read some bytes.